numpy/ma/core.py

```diff
@@ -2835,6 +2835,9 @@ def __new__(cls, data=None, mask=nomask, dtype=None, copy=False,
         # Process mask.
         # Type of the mask
         mdtype = make_mask_descr(_data.dtype)
+        
+        if mask is None:
+            mask = False
```

Comment:
Are you up for following-up?  I would slightly prefer if you add the check at the bottom in the `else:` clause.  Otherwise it is much less obvious that this does not hit the nomask branch.
A brief comment saying that None will cast to bool anyway but may be a common 
"mistake" might be nice also.

We probably don't have a test for it, in that sense it would be nice to add one just because we always add a test.  Currently None only works implicitly (because it can be converted to a bool).

Reply:
Thanks for the feedback!
I moved the code to the `else:` clause. 
I also wrote a test in the corresponding file `numpy/ma/tests/test_core.py`.